### PR TITLE
Setup CodeQL

### LINKED ISSUE #766: Setup CodeQL
### Description

Setup github actions CodeQL for all PR to improve the code quality.

### Motivation

- improve the code quality
- static analysis

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?

This PR resolves #766

### How was it solved?

:seedling: Add codeql

### How was it tested?

Manually

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# supported CodeQL languages.
+#
+name: 'CodeQL'
+
+on:
+  push:
+    branches: ['development', feature/*, hotfix, main, release/*]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ['development']
```
Comment (sameersubudhi) [inline .github/workflows/codeql.yml:19]: @ManuGowda  Should we extend the CodeQL to run against PRs based from `feature/*`and `release/*` branches as well?

Comment (ManuGowda) [reply .github/workflows/codeql.yml:19]: i just took the template 😅 , so we can extend this configuration

Comment (sameersubudhi) [reply .github/workflows/codeql.yml:19]: That's only when we `push` the changes (or merge PRs into) those specific branches but not on PRs that were branched out on them. For now, as per my understanding, it'll only run on PRs branched out from the `development` branch.

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: 'CodeQL'
+
+on:
+  push:
+    branches: ['development', feature/*, hotfix, main, release/*]
```
Comment (sameersubudhi) [inline .github/workflows/codeql.yml:16]: Another minor change to be considered.
```suggestion
    branches: ['development', feature/*, hotfix/*, main, release/*]
```

**Review by sameersubudhi — APPROVED** [review]
Looks good overall but we'll still need the following change! 🙂

**Review by nagdahimanshu — APPROVED** [review]

**Pull request merged**